connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilder.java

```diff
@@ -0,0 +1,275 @@
+     * our internal users model.
+     */
+    @Inject
+    ITResponseProcessor itResponseProcessor;
+
+    /**
+     * Processor which builds the payload that BOP/MBOP expects.
+     */
+    @Inject
+    ITUserRequestPreparer itUserRequestPreparer;
+
+    /**
+     * Predicate which decides if we should keep looping for more user pages.
+     */
+    @Inject
+    NotFinishedFetchingAllPages notFinishedFetchingAllPages;
+
+    /**
+     * Processor which transforms the RBAC response into our internal users
+     * model.
+     */
+    @Inject
+    RBACUsersProcessor rbacUsersProcessor;
+
+    /**
+     * Processor which filters the users that should not be notified for a
+     * given event.
+     */
+    @Inject
+    RecipientsFilter recipientsFilter;
+
+    /**
+     * Template processor which takes the templates and the event and
+     * transforms them to a rendered string.
+     */
+    @Inject
+    TemplateProcessor templateProcessor;
+
+    /**
+     * Aggregator which aggregates all the users from the split into a single
+     * exchange property for later to be used.
+     */
+    @Inject
+    UsersAggregationStrategy usersAggregationStrategy;
+
+    /**
+     * Processor which extracts the "is the group platform default?" key from
+     * the received payload, to determine whether it is needed to fetch the
+     * principals from that group or just all the principals from the tenant.
+     */
+    @Inject
+    RBACGroupProcessor rbacGroupProcessor;
+
+    @Override
+    public void configureRoute() {
+        from(direct(ENGINE_TO_CONNECTOR))
+            .routeId(this.connectorConfig.getConnectorName())
+            .to(direct(Routes.FETCH_AND_FILTER_RECIPIENTS));
+
+        /*
+         * Dispatches the exchanges either to fetch the users directly, or, if
+         * a group ID is present, fetch the user from that group from RBAC.
+         * After the users from the user providers have been fetched, it
+         * aggregates them and moves to the "render templates" step.
+         */
+        from(direct(Routes.FETCH_AND_FILTER_RECIPIENTS))
+            .split((simpleF("${exchangeProperty.%s}", Properties.RECIPIENT_SETTINGS)), this.usersAggregationStrategy)
+                // Store the current exchange in a property, in order to be
+                // able to access it easily later.
+                .setProperty(Properties.CURRENT_RECIPIENT_SETTINGS, body())
+                .choice()
+                    .when(simpleF("${exchangeProperty.%s.groupUUID} == null", Properties.CURRENT_RECIPIENT_SETTINGS))
+                        .to(direct(Routes.FETCH_USERS))
+                    .endChoice()
+                    .otherwise()
+                        .log(INFO, this.getClass().getName(), "Togrouphehee")
+                        .to(direct(Routes.FETCH_GROUP))
+                    .endChoice()
+                .end()
+            .end()
+            .to(direct(Routes.RENDER_TEMPLATES));
```

Comment:
Not mandatory: did you try writing that kind of code inside a `Processor`? It can make things simpler because you can use plain old Java rather than Camel methods.

```
@Inject
MyProcessor myProcessor
```

```
from(direct(Routes.FETCH_AND_FILTER_RECIPIENTS))
    .process(myProcessor);
```

```
public class MyProcessor implements Processor {

    @Inject
    ProducerTemplate producerTemplate;

    @Override
    public void process(Exchange exchange) throws Exception {
        // Do whatever you need to do here. Plain old Java is available!
        producerTemplate.send("direct:" + Routes.RENDER_TEMPLATES, exchange);
    }
}
```

Reply:
I'm going to try that. It might simplify the routes. Thanks!

Reply:
Refactored!